src/composables/use-related.js

```diff
@@ -0,0 +1,26 @@
+import AudioService from '~/data/audio-service'
+import ImageService from '~/data/image-service'
+import { ref, useFetch } from '@nuxtjs/composition-api'
+import { AUDIO, IMAGE } from '~/constants/media'
+
+const services = { [AUDIO]: AudioService, [IMAGE]: ImageService }
+
+export default function useRelated({
+  mediaType,
+  mediaId,
+  service = services[mediaType],
+}) {
+  const media = ref([])
+  // fetch and fetchState are available as this.$fetch and this.$fetchState
+  // in components, so there's no need to export them,
+  // see https://composition-api.nuxtjs.org/lifecycle/usefetch/
+  // eslint-disable-next-line no-unused-vars
+  const { fetch, fetchState } = useFetch(async () => {
+    const response = await service.getRelatedMedia({
+      id: mediaId.value,
+    })
+    media.value = response.data.results
+  })
+  fetch()
```

Comment:
```suggestion
  useFetch(async () => {
    const response = await service.getRelatedMedia({
      id: mediaId.value,
    })
    media.value = response.data.results
  }).fetch()
```

Perhaps _this_ style would look less weird, for hooks that don't return anything and are just being called? it does feel extremely weird for `fetch` and `fetchState` not to be returned and not declaring them as vars might help ease that confusion.

cc @sarayourfriend is this an awful idea lol?

Reply:
The linked documentation shows something weird:

https://composition-api.nuxtjs.org/lifecycle/usefetch/

```vue
import { defineComponent, ref, useFetch } from '@nuxtjs/composition-api'
import axios from 'axios'

export default defineComponent({
  setup() {
    const name = ref('')

    const { fetch, fetchState } = useFetch(async () => {
      name.value = await axios.get('https://myapi.com/name')
    })

    // Manually trigger a refetch
    fetch()

    // Access fetch error, pending and timestamp
    fetchState

    return { name }
  },
})

```

What's with fetchState not being called? Is that just a typo?

Reply:
`fetchState` is an object I believe... I think the example you shared is just indicating that you _can do something_ with it there.

I think it's fine to destructure `fetch` and call it separately... I don't have a preference there really (it's probably easier to read as seprate lines?) 

We should avoid destructuring unused variables though (like `fetchState`) and just throw them out completely... There's no need to get a reference to it anyway so I think it's better to just do `const { fetch } = ...` instead.

Reply:
You're right, `fetchState` is an object so that must just be an example. Thanks for the suggestion and I've updated my _code_ suggestion to remove `fetchState` from the destructuring assignment.